src/Schema/Directives/BaseDirective.php

```diff
@@ -76,19 +87,34 @@ public function getResolverFromArgument(string $argumentName): Closure
     }
 
     /**
-     * Does the current directive have an argument with the given name?
+     * Loads directive argument values from AST and caches them in $directiveArgs.
      */
-    public function directiveHasArgument(string $name): bool
+    protected function loadArgValues(): void
     {
-        return ASTHelper::directiveHasArgument($this->directiveNode, $name);
+        $this->directiveArgs = [];
+        foreach ($this->directiveNode->arguments as $node) {
+            if (! property_exists($node, 'name')) {
+                throw new Exception('Expected a Node with a name property, got: '.get_class($node));
+            }
+
+            if (array_key_exists($node->name->value, $this->directiveArgs)) {
+                continue;
```

Comment:
I'm not sure about this check.

It ensures that if there is more than one argument with the same name, the first one will be taken (as it does now). It makes no sense to specify two arguments with the same name, but it works now and GraphQL specification doesn't forbid this explicitly (at least I haven't found it). 

I can add a test to check this case.

Reply:
This should throw a `DefinitionException`. A test for this would be great.